### PR TITLE
feat(presentForm): wire into General role + markdown submit (#826)

### PR DESCRIPTION
**Pull request by isamu**
## Summary

- Make the existing `presentForm` plugin available in the **General** role with a prompt nudge so the LLM uses it for choice / yes-no / feature-toggle prompts.
- Fork `@mulmochat-plugin/form` into `src/plugins/presentForm/` and change the form-submit text from a JSON-wrapped object to a human-readable markdown bullet list (`- {label}: {value}`).
- Drop the `@mulmochat-plugin/form` npm dependency.

This is the smaller alternative to building a separate `askUserChoice` plugin (the original direction in plan PR #827). `presentForm` already covers the use case — it just wasn't wired into General and emitted JSON instead of markdown.

## Items to Confirm / Review

- [ ] The General role's new prompt section is clear enough that the LLM consistently picks `presentForm` for choice questions (manual test: ask Claude `色は red / blue / green のどれにしますか?` → expect a form, not prose).
- [ ] The markdown submission format is parseable by the LLM in practice. The plan called for plain bullets; I added a bold form-title header line if the form has a title — confirm that doesn't confuse downstream parsing.
- [ ] The fork in `src/plugins/presentForm/` was lint-cleaned (i18n + cognitive-complexity splits + slow-regex fix). `getErrorMessage` is now a dispatcher across 5 small validators — verify this didn't change validation behavior. Email validation switched from regex `/^[^\s@]+@[^\s@]+\.[^\s@]+$/` to indexOf-based parsing (sonarjs flagged the regex as backtracking-prone); behavior should match common cases but worth a glance.
- [ ] 7 non-English locales (`ja/zh/ko/es/fr/de/pt-BR`) translated freshly — none are pasted English fallbacks.

## User Prompt

> LLMがわからどれにしますか?選択してください、みたいな返答のときに、checkboxやselectのUIで選択して返信できると楽だね。案作れる?checkboxやselect以外もあっても良いかもしれないし、質問は複数あるかもしれない。

Plan PR #827 (merged 2026-04-26) described building a new `askUserChoice` MCP tool. While starting the implementation, I found that the existing `presentForm` plugin (already in `office`/`guide`/`tutor` roles) covers the same ground with a richer field-type set. After confirming with the user that this counted as a duplicate, the work pivoted to:

> あれ、じゃぁpresentFormをgeneralにいれてみて
> a b はやろう。そのうえでPR化。
> あー、~/ss/llm/GUIChatPlugins/MulmoChatPluginForm/にあるからコピーして使おう

— wire `presentForm` into General, copy the upstream source so we can change the submit format to markdown, and add a prompt instruction to mark fields as `required: true`.

### General role
- `src/config/roles.ts` — `presentForm` added to `availablePlugins`. New "Asking the user to choose" section in the prompt covers when to use the form and the `required: true` instruction.

### Local fork of @mulmochat-plugin/form
- `src/plugins/presentForm/{types,definition,plugin,index}.ts` + `{View,Preview}.vue` — copied from `~/ss/llm/GUIChatPlugins/MulmoChatPluginForm/` (snakajima + isamu's source repo).
- `View.vue:handleSubmit` — emits a markdown bullet list instead of `{"formSubmission":{...}}` JSON. Empty values render as `(none)`.
- Compliance refactors in the fork: i18n-ed all raw template text, split `getErrorMessage` into 5 per-type validators to drop cognitive complexity, replaced the email regex with indexOf-based parsing (sonarjs slow-regex), renamed single-letter `f` → `field`, fixed import paths from `../core/types` → `./types`.

### Wiring
- `src/tools/index.ts` — swap `FormPlugin.plugin` (npm) → local `presentFormPlugin`.
- `server/agent/plugin-names.ts` + `server/api/routes/plugins.ts` — server-side imports point at the local fork.
- `package.json` (root) + `packages/mulmoclaude/package.json` + `yarn.lock` — `@mulmochat-plugin/form` removed.

### i18n
- `pluginPresentForm` block added to all 8 locale files (`en/ja/zh/ko/es/fr/de/pt-BR`).

## Test plan

- [ ] Restart `yarn dev` (server-side files changed). General role.
- [ ] Ask `色は red / blue / green のどれにしますか?` — expect a form in the canvas with a radio group, not prose.
- [ ] Submit `blue` → chat shows `- 色: blue` style markdown, not JSON. The LLM reads it and continues with the answer.
- [ ] Multi-question: `色 + 機能 (検索/通知/同期 複数可) を聞いて` — one form with radio + checkbox.
- [ ] Required gating: a `required:true` field blocks Submit until filled.
- [ ] Roles other than General that already had `presentForm` (`office`/`guide`/`tutor`) — verify still working.

## Out of scope

- Moving `plans/feat-ask-user-choice.md` to `plans/done/` — handled by `pr-merge-tidy` after this PR merges.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/plugins/presentForm/View.vue [hunk 1]
```diff
@@ -0,0 +1,640 @@
...
+    return;
+  }
+
+  // Build a markdown bullet list of `- {label}: {value}` lines so the
+  // chat history reads naturally for a human while still being trivial
+  // for the LLM to parse. Empty values become "(none)".
+  const lines: string[] = [];
+  if (formData.value?.title) lines.push(`**${formData.value.title}**`, "");
+  formData.value?.fields.forEach((field) => {
+    lines.push(`- ${field.label}: ${renderValue(field, formValues.value[field.id])}`);
```
Comment (isamu) [inline src/plugins/presentForm/View.vue]: This writes raw labels/values directly into markdown. A textarea value with newlines, or a label/value containing `- `, `:`, `*`, links, etc. can split the bullet list or inject extra pseudo-bullets, which makes the LLM-facing payload ambiguous. Please escape or encode the fields before composing this message.

File: src/plugins/presentForm/View.vue [hunk 2]
```diff
@@ -0,0 +1,640 @@
...
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+function renderValue(field: FormField, value: any): string {
+  const empty = "(none)";
+  if (field.type === "radio" || field.type === "dropdown") {
+    return value !== null && value !== undefined ? field.choices[value] : empty;
+  }
+  if (field.type === "checkbox") {
+    const items = (value || []).map((idx: number) => field.choices[idx]);
+    return items.length > 0 ? items.join(", ") : empty;
```
Comment (isamu) [inline src/plugins/presentForm/View.vue]: Joining checkbox selections with commas is ambiguous once a choice itself contains a comma. That means the markdown no longer round-trips reliably for the LLM. Please use an unambiguous separator/encoding per selection or escape commas in the rendered values.

File: src/plugins/presentForm/plugin.ts [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    throw new Error(`Field '${field.id}': min cannot be greater than max`);
+  }
+  if (field.type === "date" && field.minDate && field.maxDate && field.minDate > field.maxDate) {
+    throw new Error(`Field '${field.id}': minDate cannot be after maxDate`);
+  }
+  if (field.type === "checkbox") {
+    if (field.minSelections !== undefined && field.maxSelections !== undefined && field.minSelections > field.maxSelections) {
+      throw new Error(`Field '${field.id}': minSelections cannot be greater than maxSelections`);
+    }
+    if (field.maxSelections !== undefined && field.maxSelections > field.choices.length) {
```
Comment (isamu) [inline src/plugins/presentForm/plugin.ts]: This only checks `maxSelections` against the number of choices. A form with `minSelections > choices.length` will still validate here, but it can never be submitted successfully. Please add the missing lower-bound check as well.

**Review by isamu — COMMENTED** [review]
Three issues worth addressing: markdown serialization is brittle for free-text/markdown-special inputs, checkbox selection rendering is ambiguous for comma-containing options, and checkbox validation still allows impossible configurations.

**Comment by isamu** [comment]
CODEX VERDICT: CHANGES REQUESTED

- `src/plugins/presentForm/View.vue:handleSubmit` writes raw labels/values into markdown, so newline or markdown-special characters can break the bullet format and create parsing/prompt-injection ambiguity.
- Checkbox submission rendering joins selections with commas, so comma-containing choices are ambiguous on the way back through chat.
- `src/plugins/presentForm/plugin.ts` still allows `minSelections > choices.length`, which produces an impossible form state.
- There is no direct test coverage for the new markdown serialization path or the validation boundary cases above.

**Comment by isamu** [comment]
CODEX VERDICT: CHANGES REQUESTED

- `src/plugins/presentForm/View.vue:590-629` only addresses multiline `text`/`textarea` wrapping. It still injects raw `formData.title`, `field.label`, and checkbox/radio choice text into markdown, so characters like `*`, `- `, `:`, links, or newlines in labels/choices can still reshape the bullet structure and make the LLM-facing payload ambiguous. The serialization fix is therefore only partial.
- `src/plugins/presentForm/plugin.ts:26-38` looks fine: the new `minSelections > choices.length` guard is present, and the existing `minSelections > maxSelections` / `maxSelections > choices.length` checks are still in place after the destructuring refactor.

**Comment by isamu** [comment]
## Codex cross-review converged — LGTM after iteration 3

**Iteration 1** (commit `fd7d617f`)
Codex flagged: (a) markdown brittleness on text/textarea newlines, (b) checkbox comma-join ambiguity, (c) missing `minSelections > choices.length` validation. All three addressed.

**Iteration 2** (commit `29207ccb`)
Codex flagged: title/label/choice strings were still interpolated raw — newlines in those (LLM-authored) fields could reshape the bullet structure. Added `singleLine()` helper applied at three injection sites; multi-line free-form input (text/textarea values) kept as `indentContinuation()`.

**Iteration 3** — Codex CLI output:
> The `singleLine()` fix covers the remaining markdown-shaping path: the form title, every field label, and every radio/dropdown/checkbox choice are normalized before interpolation, while free-form text and textarea values still preserve multiline input via `indentContinuation()`.
>
> No outstanding concerns from the prior finding.
>
> `CODEX VERDICT: LGTM`

(Codex's own attempt to post this comment was blocked by its connector's safety filter — repeating it here for the record. Local logs at `/tmp/codex-cross-review-845/codex-iter3.log`.)

Waiting on CI before requesting merge.
